Shared/Application/Error/Handlers/TransientExceptionHandler.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Confluent.Kafka;
+using LantanaGroup.Link.Shared.Application.Error.Exceptions;
+using LantanaGroup.Link.Shared.Application.Error.Interfaces;
+using LantanaGroup.Link.Shared.Application.Interfaces;
+using LantanaGroup.Link.Shared.Application.Models;
+using LantanaGroup.Link.Shared.Application.Models.Kafka;
+using Microsoft.Extensions.Logging;
+
+namespace LantanaGroup.Link.Shared.Application.Error.Handlers
+{
+    public class TransientExceptionHandler<K, V> : ITransientExceptionHandler<K, V>
+    {
+        protected readonly ILogger<TransientExceptionHandler<K, V>> Logger;
+        protected readonly IKafkaProducerFactory<string, AuditEventMessage> AuditProducerFactory;
+        protected readonly IKafkaProducerFactory<K, V> ProducerFactory;
+
+        /// <summary>
+        /// The Topic to use when publishing Retry Kafka events.
+        /// </summary>
+        public string Topic { get; set; } = string.Empty;
+
+        /// <summary>
+        /// The name of the service that is consuming the TransientExceptionHandler.
+        /// </summary>
+        public string ServiceName { get; set; } = string.Empty;
+
+        protected TransientExceptionHandler(ILogger<TransientExceptionHandler<K, V>> logger,
+            IKafkaProducerFactory<string, AuditEventMessage> auditProducerFactory,
+            IKafkaProducerFactory<K, V> producerFactory)
+        {
+            Logger = logger;
+            AuditProducerFactory = auditProducerFactory;
+            ProducerFactory = producerFactory;
+        }
+
+        public virtual void HandleException(ConsumeResult<K, V> consumeResult, Exception ex)
```

Comment:
Would it make sense to pass the topic name as a method argument here and pass it down as opposed to setting it in the calling class?